c_src/erlzmq_nif.c

```diff
@@ -877,6 +881,74 @@ SOCKET_COMMAND(erlzmq_socket_command_send)
   return result;
 }
 
+SOCKET_COMMAND(erlzmq_socket_command_send_multipart)
+{
+  assert(argc == 2);
+  int flags;
+
+  assert(socket->socket_zmq);
+
+  unsigned n;
+  if (! enif_get_list_length(env, argv[0], &n) || n == 0) {
+    return enif_make_badarg(env);
+  }
+
+  if (! enif_get_int(env, argv[1], &flags)) {
+    return enif_make_badarg(env);
+  }
+
+  ERL_NIF_TERM result = enif_make_atom(env, "ok");
+  ERL_NIF_TERM head, tail;
+
+  int initialized_messages = 0;
+  zmq_msg_t *msg = enif_alloc(n * sizeof(zmq_msg_t));
+  if (! msg) {
+    result = return_zmq_errno(env, ENOMEM);
+    goto cleanup;
+  }
+
+  enif_get_list_cell(env, argv[0], &head, &tail);
+  for (int i = 0; i<n; i++) {
+    ErlNifBinary binary;
+    if (! enif_inspect_binary(env, head, &binary)) {
+      result = enif_make_badarg(env);
+      goto cleanup;
+    }
+    if (zmq_msg_init_size(&msg[i], binary.size)) {
+      result = return_zmq_errno(env, zmq_errno());
+      goto cleanup;
+    }
+    initialized_messages++;
+    void * data = zmq_msg_data(&msg[i]);
+    if (! data) {
+      result = return_zmq_errno(env, ENOMEM);
+      goto cleanup;
+    }
+    memcpy(data, binary.data, binary.size);
+    enif_get_list_cell(env, tail, &head, &tail);
+  }
+
+  for (int i = 0; i < n;) {
+    int sndmore = (i < n-1) ? ZMQ_SNDMORE : 0;
+    if (zmq_sendmsg(socket->socket_zmq, &msg[i], flags|sndmore) == -1) {
+      if (errno == EINTR && i>0)
```

Comment:
use zmq_errno

Reply:
Do you have an idea if we can easily test that?

Reply:
This code is copied from the czmq library, which uses errno for the exact same test. Will have to verify that zmq_errno is right thing to use here.

With regard to testing: no, I don't know how to test this. But as I said: the code is basically equivalent to what czmq does, and I use this code ro process 5 billion messages per day, so there's a high chance it's correct.

Reply:
done